DDInstall/DDInstall.psm1

```diff
@@ -0,0 +1,192 @@
+function Install-DDAgent
```

Comment:
Since Github doesn't seem to know how to display `psd1` files, I'm copying the important parts here for visibility:
```
# Script module or binary module file associated with this manifest.
RootModule = 'DDInstall.psm1'

# Version number of this module.
ModuleVersion = '1.0'

# Supported PSEditions
# CompatiblePSEditions = @()

# ID used to uniquely identify this module
GUID = 'd8ed3d7e-ed2f-4e84-9987-02fcfa3aed19'

# Company or vendor of this module
CompanyName = 'Datadog'

# Copyright statement for this module
Copyright = '(c) 2024 Datadog. All rights reserved.'

# Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
FunctionsToExport = @("Install-DDAgent")
```

One part I think is important to discuss is the module's name - I used `DDInstall` as a placeholder, but I don't love it. @richl-dd, any suggestions?

Reply:
How does everyone feel about just calling it Datadog? I looked through the most popular packages (https://www.powershellgallery.com/packages) and the convention seems to be either a single name (e.g. `Datadog`) or something like `CompanyName.Thing.SubThing` (e.g. `Microsoft.WinGet.Client`).

I think `Install-Module Datadog` would be fine - thoughts (@shanelhuang @ISauve @clarkb7 @derekwbrown)?

Reply:
Works for me!